cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/FileSetTest.java

```diff
@@ -34,33 +40,56 @@
 
 public class FileSetTest extends AbstractDatasetTest {
 
-  static FileSet fileSet;
-  private static final Id.DatasetInstance testFileSetInstance = Id.DatasetInstance.from(NAMESPACE_ID, "testFileSet");
+  static FileSet fileSet1;
+  static FileSet fileSet2;
+  private static final Id.Namespace NAMESPACE2 = Id.Namespace.from("yourspace");
+  private static final DatasetNamespace dsNamespace = new DefaultDatasetNamespace(CConfiguration.create());
+  private static final Id.DatasetInstance testFileSetInstance1 =
+    dsNamespace.namespace(Id.DatasetInstance.from(NAMESPACE_ID, "testFileSet"));
+  private static final Id.DatasetInstance testFileSetInstance2 =
+    dsNamespace.namespace(Id.DatasetInstance.from(NAMESPACE2, "testFileSet"));
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    createInstance("fileSet", testFileSetInstance, FileSetProperties.builder()
+    createInstance("fileSet", testFileSetInstance1, FileSetProperties.builder()
       .setBasePath("testDir").build());
     Map<String, String> fileArgs = Maps.newHashMap();
     FileSetArguments.setInputPath(fileArgs, "some?File");
     FileSetArguments.setOutputPath(fileArgs, "some?File");
-    fileSet = getInstance(testFileSetInstance, fileArgs);
+    fileSet1 = getInstance(testFileSetInstance1, fileArgs);
+
+    framework.addModule(Id.DatasetModule.from(NAMESPACE2, "fileSet"), new FileSetModule());
+    createInstance("fileSet", testFileSetInstance2, FileSetProperties.builder()
+      .setBasePath("testDir").build());
+    fileArgs = Maps.newHashMap();
+    FileSetArguments.setInputPath(fileArgs, "some?File");
+    FileSetArguments.setOutputPath(fileArgs, "some?File");
+    fileSet2 = getInstance(testFileSetInstance2, fileArgs);
   }
 
   @AfterClass
   public static void afterClass() throws Exception {
-    deleteInstance(testFileSetInstance);
+    deleteInstance(testFileSetInstance1);
+    deleteInstance(testFileSetInstance2);
   }
 
   @Test
   public void testWriteRead() throws IOException {
-    OutputStream out = fileSet.getOutputLocation().getOutputStream();
+    Location fileSet1Location = fileSet1.getOutputLocation();
+    Location fileSet2Location = fileSet2.getOutputLocation();
+    Location fileSet1NsDir = Locations.getParent(Locations.getParent(Locations.getParent(fileSet1Location)));
+    Location fileSet2NsDir = Locations.getParent(Locations.getParent(Locations.getParent(fileSet2Location)));
+    Assert.assertNotNull(fileSet1NsDir);
+    Assert.assertNotNull(fileSet2NsDir);
+    Assert.assertEquals(fileSet1NsDir.getName(), NAMESPACE_ID.getId());
+    Assert.assertEquals(fileSet2NsDir.getName(), NAMESPACE2.getId());
+
+    OutputStream out = fileSet1.getOutputLocation().getOutputStream();
     out.write(42);
     out.close();
 
-    InputStream in = fileSet.getInputLocations().get(0).getInputStream();
+    InputStream in = fileSet1.getInputLocations().get(0).getInputStream();
     Assert.assertEquals(42, in.read());
     in.close();
```

Comment:
don't you want to assert that the two filesets don't see each other's files? Or that the locations are different?